src/main/java/org/candlepin/subscriptions/http/HttpClient.java

```diff
@@ -68,4 +82,40 @@ public static Client buildHttpClient(HttpClientProperties serviceProperties, Obj
 
         return ((ResteasyClientBuilder) clientBuilder).httpEngine(engine).build();
     }
+
+    private static SSLContext getSslContextFromKeystore(HttpClientProperties serviceProperties) {
+        final File keyStoreFile = serviceProperties.getKeystore();
+        final char[] keyStorePassword;
+        if (serviceProperties.getKeystorePassword() == null) {
+            keyStorePassword = "".toCharArray();
+        }
+        else {
+            keyStorePassword = serviceProperties.getKeystorePassword();
+        }
+
+        if (keyStoreFile.exists() && keyStoreFile.canRead()) {
+            try (final BufferedInputStream bufferedInputStream =
+                new BufferedInputStream(new FileInputStream(keyStoreFile))) {
+                final KeyStore store = KeyStore.getInstance("JKS");
+                store.load(bufferedInputStream, keyStorePassword);
+                final KeyManagerFactory kmf =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                kmf.init(store, keyStorePassword);
+                final TrustManagerFactory tmf =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                tmf.init((KeyStore) null);
+                final SSLContext ctx = SSLContext.getInstance("TLS");
+                ctx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+                return ctx;
+            }
+            catch (IOException | GeneralSecurityException e) {
+                throw new IllegalStateException(String.format("Keystore file %s could not be accessed! %s",
+                    keyStoreFile.getAbsolutePath(), e.getMessage()));
+            }
+        }
+        else {
+            throw new IllegalStateException(String.format("Keystore file %s is not visible!",
+                keyStoreFile.getAbsolutePath()));
+        }
+    }
```

Comment:
@awood can you please spot check this method specifically? I'm not as familiar with SSL context initialization as you are :-)